### PR TITLE
Implements printing functionality

### LINKED ISSUE #13: Introduce a way to print to the screen
We need to be able to print stuff to the screen.
This would initially only be used for debug messages, and would not need all the functions necessary for the full project.

There is a great chapter on this on [this rust tutorial](https://os.phil-opp.com/vga-text-mode/).

### PR DESCRIPTION
**Pull request by Elekrisk**
`println!` can now be used to print to the screen,
after the printer has been initialized.

Co-authored-by: brynolv <k.jzfgxgh@gmail.com>
Co-authored-by: Vikdos <viktor.ofjall@gmail.com>

Fixes #13

### CODE REVIEW

File: src/printer.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            font: &DEFAULT_FONT
+        }
+    }
+
+    unsafe fn clear(&mut self) {
+        let res_y = self.framebuffer.info().vertical_resolution;
+        let res_x = self.framebuffer.info().horizontal_resolution;
+        let stride = self.framebuffer.info().stride;
+        let bytes_per_pixel = self.framebuffer.info().bytes_per_pixel;
+        let buffer = self.framebuffer.buffer_mut();
```
Comment (brynolv) [inline src/printer.rs:29]: Like here

File: src/printer.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        match char {
+            '\n' => {
+                cursor_y += 1;
+                cursor_x = 0;
+            },
+            other if other < ' ' || other == '\x7F' => {},
+            _ => {
+                let stride = self.framebuffer.info().stride;
+                let bytes_per_pixel = self.framebuffer.info().bytes_per_pixel;
+                let buffer = self.framebuffer.buffer_mut();
```
Comment (brynolv) [inline src/printer.rs:53]: And here...

File: src/printer.rs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                for b in 0..bytes_per_pixel {
+                    buffer[(y * stride + x) * bytes_per_pixel + b] = 0;
+                }
+            }
+        }
+    }
+
+    fn get_buffer_info(&mut self) -> (usize, usize, usize, usize, &mut [u8]) {
+        (self.framebuffer.info().horizontal_resolution, self.framebuffer.info().vertical_resolution, self.framebuffer.info().stride, self.framebuffer.info().bytes_per_pixel, self.framebuffer.buffer_mut())
+    }
```
Comment (brynolv) [inline src/printer.rs:100]: So more things like this 😃

**Review by Vikdos — APPROVED** [review]

**Review by brynolv — APPROVED** [review]
I'm going to look into prettifying/refactoring later, for now good to go!

**Pull request merged**